src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -327,7 +328,8 @@ protected [db] class CampaignStatusTransition(implicit db: Database, ec: Executi
       }
     } yield ()
 
-  protected [db] def calculateCampaignStatus(campaign: CampaignId): DBIO[Either[Unit, CampaignStatus]] = {
+  protected [db] def calculateCampaignStatus(
+      campaign: CampaignId, currentCampaignStatus: Option[CampaignStatus] = None): DBIO[Either[Unit, CampaignStatus]] = {
```

Comment:
Having `DBIO[Either[Unit, CampaignStatus]]` as return type feels odd... Could this be `DBIO[Option[CampaignStatus]]`?

Reply:
I guess so.